src/canister/individual_user_template/src/api/hot_or_not_bet/receive_bet_from_bet_makers_canister.rs

```diff
@@ -80,6 +90,196 @@ fn receive_bet_from_bet_makers_canister_impl(
+fn receive_bet_from_bet_makers_canister_v1(
+    place_bet_arg: PlaceBetArg,
+    bet_maker_principal_id: Principal,
+) -> Result<BettingStatusV1, BetOnCurrentlyViewingPostError> {
+    let bet_maker_canister_id = ic_cdk::caller();
+    update_last_canister_functionality_access_time();
+    let current_time = system_time::get_current_system_time_from_ic();
+
+    let status = CANISTER_DATA.with(|canister_data_ref_cell| {
+        let canister_data = &mut canister_data_ref_cell.borrow_mut();
+
+        receive_bet_from_bet_makers_canister_impl_v1(
+            canister_data,
+            &bet_maker_principal_id,
+            &bet_maker_canister_id,
+            place_bet_arg.clone(),
+            &system_time::get_current_system_time_from_ic(),
+        )
+    })?;
+
+    CANISTER_DATA.with(|canister_data_ref_cell| {
+        maybe_start_timer_based_on_bet_result(
+            &mut canister_data_ref_cell.borrow_mut(),
+            current_time,
+            place_bet_arg.clone(),
+            status.clone(),
+        );
+    });
+
+    CANISTER_DATA.with(|canister_data_ref_cell| {
+        update_profile_stats_with_bet_placed(
+            &mut canister_data_ref_cell.borrow_mut(),
+            &place_bet_arg.bet_direction,
+        );
+    });
+
+    update_scores_and_share_with_post_cache_if_difference_beyond_threshold(&place_bet_arg.post_id);
+
+    Ok(status)
+}
+
+fn receive_bet_from_bet_makers_canister_impl_v1(
+    canister_data: &mut CanisterData,
+    bet_maker_principal_id: &Principal,
+    bet_maker_canister_id: &CanisterId,
+    place_bet_arg: PlaceBetArg,
+    current_time: &SystemTime,
+) -> Result<BettingStatusV1, BetOnCurrentlyViewingPostError> {
+    let PlaceBetArg {
+        post_id,
+        bet_amount,
+        bet_direction,
+        ..
+    } = place_bet_arg;
+
+    let post = canister_data.all_created_posts.get_mut(&post_id).unwrap();
+
+    post.place_hot_or_not_bet_v2(
+        bet_maker_principal_id,
+        bet_maker_canister_id,
+        bet_amount,
+        &bet_direction,
+        current_time,
+        &mut canister_data.room_details_map_v1,
+        &mut canister_data.bet_details_map_v1,
+        &mut canister_data.post_principal_map,
+        &mut canister_data.slot_details_map_v1,
+    )
+}
+
+fn maybe_start_timer_based_on_bet_result(
+    canister_data: &mut CanisterData,
+    current_time: SystemTime,
+    place_bet_arg: PlaceBetArg,
+    bet_result: BettingStatusV1,
+) {
+    // if let Ok(ok_bet_result) = bet_result {
+    match bet_result {
+        // this case should never match in yral game implementation
+        BettingStatusV1::BettingClosed => {
+            ();
+        }
+        BettingStatusV1::BettingOpen {
+            ongoing_slot,
+            ongoing_room,
+            ..
+        } => {
+            // insert only the first bet in first_bet_placed_at_hashmap
+            if !canister_data
+                .first_bet_placed_at_hashmap
+                .contains_key(&place_bet_arg.post_id)
```

Comment:
We should also compare the ongoing slot for in this check as if we the ongoing slot is different and we miss inserting that into the queue it would never be calculated.